powerauth-test-server/src/main/java/com/wultra/security/powerauth/app/testserver/service/SignatureService.java

```diff
@@ -0,0 +1,124 @@
+import com.wultra.security.powerauth.app.testserver.database.entity.TestConfigEntity;
+import com.wultra.security.powerauth.app.testserver.errorhandling.ActivationFailedException;
+import com.wultra.security.powerauth.app.testserver.errorhandling.AppConfigNotFoundException;
+import com.wultra.security.powerauth.app.testserver.errorhandling.RemoteExecutionException;
+import com.wultra.security.powerauth.app.testserver.model.request.ComputeOnlineSignatureRequest;
+import com.wultra.security.powerauth.app.testserver.model.response.ComputeOnlineSignatureResponse;
+import com.wultra.security.powerauth.app.testserver.util.StepItemLogger;
+import io.getlime.security.powerauth.crypto.lib.enums.PowerAuthSignatureTypes;
+import io.getlime.security.powerauth.lib.cmd.logging.ObjectStepLogger;
+import io.getlime.security.powerauth.lib.cmd.logging.model.StepItem;
+import io.getlime.security.powerauth.lib.cmd.steps.VerifySignatureStep;
+import io.getlime.security.powerauth.lib.cmd.steps.model.VerifySignatureStepModel;
+import org.json.simple.JSONObject;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.Map;
+
+/**
+ * Service for calculating PowerAuth signatures.
+ *
+ * @author Roman Strobl, roman.strobl@wultra.com
+ */
+@Service
+public class SignatureService extends BaseService {
+
+    private final static Logger logger = LoggerFactory.getLogger(SignatureService.class);
+
+    private final TestServerConfiguration config;
+    private final ResultStatusService resultStatusUtil;
+    private final VerifySignatureStep verifySignatureStep;
+
+    /**
+     * Service constructor.
+     * @param config Test server configuration.
+     * @param resultStatusUtil Result status utilities.
+     * @param verifySignatureStep Verify signature step.
+     */
+    @Autowired
+    public SignatureService(TestServerConfiguration config, TestConfigRepository appConfigRepository, ResultStatusService resultStatusUtil, VerifySignatureStep verifySignatureStep) {
+        super(appConfigRepository);
+        this.config = config;
+        this.resultStatusUtil = resultStatusUtil;
+        this.verifySignatureStep = verifySignatureStep;
+    }
+
+    /**
+     * Compute a token digest.
+     * @param request Request for computing a token digest.
+     * @return Response for computing a token digest.
+     * @throws RemoteExecutionException In case remote communication fails.
+     * @throws ActivationFailedException In case activation is not found.
+     * @throws AppConfigNotFoundException In case application configuration is not found.
+     */
+    @SuppressWarnings("unchecked")
+    public ComputeOnlineSignatureResponse computeOnlineSignature(ComputeOnlineSignatureRequest request) throws RemoteExecutionException, ActivationFailedException, AppConfigNotFoundException {
+
+        final String applicationId = request.getApplicationId();
```

Comment:
What about to add some INFO or DEBUG log (with important context) that the method was called?

Reply:
Let's do this as part of another ticket, the log messages are missing everywhere